src/CheckboxSelector/widget/checkboxselector.js

```diff
@@ -1,459 +1,418 @@
+
+				headTr.appendChild(checkboxHeadTh);
+
+				for (var k = 0; k < this.splits.length; k++) {
+					var headTh = domConstruct.create('th', { 'class': 'mendixDataGrid_tableHead', 'width': this.splits[k].displayWidth + '%' });
+					var firstDiv = domConstruct.create('div', { 'class': 'mendixDataGrid_headContainer' }, headTh);
+					domConstruct.create('div', { 'class': 'mendixDataGrid_sortIcon' }, firstDiv);
+					domConstruct.create('div', { 'class': 'mendixDataGrid_columnCaption', 'innerHTML': this.splits[k].headers }, firstDiv);
+
+					if (k == this.splits.length - 1)
+						domClass.add(headTh, 'mendixDataGrid_tableHeadLast');
+
+					headTr.appendChild(headTh);
 				}
+				table.appendChild(thead);
 			}
-			
-			var checkboxHeadTh = mxui.dom.th(
-					{ 'class': 'mendixDataGrid_tableHead mendixDataGrid_tableHeadFirst' },
-					mxui.dom.div(
-						{ 'class': 'mendixDataGrid_headContainer' },
-						mxui.dom.div({ 'class': 'mendixDataGrid_sortIcon' }),
-						mxui.dom.div({ 'class': 'mendixDataGrid_columnCaption' }, 
-								checkall
-						)
-					)
-				);
-			
-			this.boxWidth && dojo.style( checkboxHeadTh, { width: this.boxWidth + this.boxWidthUnit });
-			
-			headTr.appendChild(checkboxHeadTh);
-			
-			for (var k = 0; k < this.splits.length; k++) {
-				var headTh = mxui.dom.th({ 'class' : 'mendixDataGrid_tableHead', 'width' : this.splits[k].displayWidth+'%'},
-								mxui.dom.div({ 'class' : 'mendixDataGrid_headContainer' },
-									mxui.dom.div( { 'class' : 'mendixDataGrid_sortIcon' }),
-									mxui.dom.div({ 'class' : 'mendixDataGrid_columnCaption' }, this.splits[k].headers)));
-				
-				if (k == this.splits.length-1)
-					mxui.dom.addClass(headTh, 'mendixDataGrid_tableHeadLast');
-				
-				headTr.appendChild(headTh);
+
+			var tbody = domConstruct.create('tbody', { 'class': 'mendixDataGrid_gridBody' });
+			table.appendChild(tbody);
+
+			this.hoverNode = null;
+			for (var i = 0; i < objs.length; i++) {
+				var bodyTr = domConstruct.create('tr', { 'class': 'mendixDataGrid_tableRow' });
+				tbody.appendChild(bodyTr);
+
+				var currObj = objs[i];
+				var currGUID = currObj.getGuid();
+				var checkBool = (array.indexOf(selectedObjs, currGUID) != -1);
+
+				var checkbox = domConstruct.create('input', { type: 'checkbox', 'class': 'checkbox_box' });
+				checkbox.checked = checkBool;
+
+				domAttr.set(checkbox, 'defaultChecked', checkBool);
+
+				this.rowsList[currGUID] = checkbox;
+
+				// DRE: Unfortunatly this is not a boolean, if the attribute is there the node will be disabled.
+				// this.isInactive && dojo.attr( checkbox, "disabled", "disabled" ); 
+
+				if (i % 2 !== 0)
+					domClass.add(bodyTr, 'mendixDataGrid_tableRowEven');
+				else
+					domClass.add(bodyTr, 'mendixDataGrid_tableRowOdd');
+
+				var checkboxTd = domConstruct.create('td', { 'class': 'mendixDataGrid_tableData' });
+				checkboxTd.appendChild(checkbox);
+
+				domStyle.set(checkboxTd, 'width', this.boxWidth + this.boxWidthUnit);
```

Comment:
Do not remove logic from the widget, especially null checks like `this.boxWidth && `

Reply:
Reverted